### PR TITLE
Add ORC support for iceberg connector

### PR DESCRIPTION
**Pull request by junyi1313**
Cherry-pick of https://github.com/trinodb/trino/pull/1067, https://github.com/trinodb/trino/pull/2042, https://github.com/trinodb/trino/pull/4055, https://github.com/trinodb/trino/pull/1629, https://github.com/trinodb/trino/pull/3483, https://github.com/trinodb/trino/commit/ecce4a23112ebff3efbe205d0a0894fbaaad997e
    
Co-authored-by: Parth Brahmbhatt <pbrahmbhatt@netflix.com>
Co-authored-by: David Phillips <david@acz.org>
Co-authored-by: Xingyuan Lin <linxingyuan1102@gmail.com>
Co-authored-by: Dain Sundstrom <dain@iq80.com>

Test plan - Unit Tests

```
== RELEASE NOTES ==

Iceberg Connector Changes
* Add ORC support for iceberg connector
```
This PR implements the issue: https://github.com/prestodb/presto/issues/16305

### CODE REVIEW

**Comment by junyi1313** [comment]
@ChunxuTang @zhenxiao @beinan Could you help take a look? Thanks.

File: presto-server/src/main/assembly/presto.xml
Comment (zhenxiao) [inline presto-server/src/main/assembly/presto.xml:189]: presto-iceberg or iceberg? either is fine, just a note

**Review by zhenxiao — APPROVED** [review]
looks good, @junyi1313 one minor thing
@beinan @ChunxuTang could you please take a look?

File: presto-server/src/main/assembly/presto.xml
Comment (junyi1313) [reply presto-server/src/main/assembly/presto.xml:189]: I think it's fine. Just like mysql connector: [assembly/presto.xml#presto-mysql](https://github.com/prestodb/presto/blob/bb6d0935dc611bd3fe87963f571a4abe8370b117/presto-server/src/main/assembly/presto.xml#L120)

**Comment by ChunxuTang** [comment]
@junyi1313 @zhenxiao 
Sure, I'm glad to help with the review. will take a look this weekend.

**Review by ChunxuTang — COMMENTED** [review]
@junyi1313 Thanks for your work! This is a very nice feature we want for the iceberg connector!!
From my initial review, the implementation on the iceberg connector part generally looks good to me. 
Just one reminder: I noticed that there're some new features/improvements but are not in the PRs you cherry-picked. I think the new features may deserve a bit more documentation or clarification as they are unique contributions.

@zhenxiao @beinan There are some changes in the presto-orc package, including the upgrade of ORC. I'm not familiar with that package. Could you folks have a closer look at those presto-orc changes? Or any other folks know more?

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergModule.java
Comment (beinan) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergModule.java:231]: I can try to provide some context of these two functions -- they are for caching the metadata of orc files -- more details: https://github.com/prestodb/presto/pull/13501/

But looks like these two are duplicated with the functions in presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java
Have we included HiveClientModule in the iceberg connector?  if not, can we reuse the functions in HiveClientModule or extract them to a separate module?

Comment (junyi1313) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergModule.java:231]: @beinan @ChunxuTang These two functions are copied from HiveClientModule because we haven't included HiveClientModule in the iceberg connector. Besides, the presto-raptor  [StorageModule](https://github.com/prestodb/presto/blob/860a1389cc304b7e649ce7c157d1c67b5e805800/presto-raptor/src/main/java/com/facebook/presto/raptor/storage/StorageModule.java#L118) also has these two functions. I think extracting them to a separate module is better. Should we do it in this PR or open a new PR?

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergOrcFileWriter.java
Comment (junyi1313) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergOrcFileWriter.java]: Fixed

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergOrcColumn.java
Comment (junyi1313) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergOrcColumn.java:1]: IcebergOrcColumn.java is similar to the [OrcColumn](https://github.com/trinodb/trino/blob/master/lib/trino-orc/src/main/java/io/trino/orc/OrcColumn.java). I have updated the cherry-pick infos and removed the unused functions.

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergModule.java
Comment (beinan) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergModule.java:231]: @junyi1313 I'm ok with either.    @ChunxuTang what do you think?

**Review by beinan — APPROVED** [review]
Looks good to me!  Great contribution, many thanks!

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergModule.java
Comment (ChunxuTang) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergModule.java:231]: @junyi1313 @beinan 
Thanks for your clarification! Yeah, I'm also ok with either way. @junyi1313 your call.

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergOrcColumn.java
Comment (ChunxuTang) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergOrcColumn.java:1]: Gotcha. Thanks for your work!

**Review by ChunxuTang — COMMENTED** [review]
@junyi1313 
Thanks for your nice work! Looks that there're some errors in CI tests. Could you fix the errors and update the PR to pass the tests?

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergModule.java
Comment (junyi1313) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergModule.java:231]: Got it. I will find time to send a new PR about this work after this PR has been merged.

**Comment by junyi1313** [comment]
@ChunxuTang I have updated the PR and fixed the CI errors. 
@zhenxiao @beinan  Pls help to merge. 
Thanks so much.

**Pull request merged**
